lib/formdata/parser.js

```diff
@@ -0,0 +1,672 @@
+    }
+
+    if (this.#buffers[0].length === n) {
+      this.#byteOffset -= this.#buffers[0].length
+      return this.#buffers.shift()
+    }
+
+    const buffer = Buffer.allocUnsafe(n)
+    let offset = 0
+
+    while (offset !== n) {
+      const next = this.#buffers[0]
+      const { length } = next
+
+      if (length + offset === n) {
+        buffer.set(this.#buffers.shift(), offset)
+        break
+      } else if (length + offset > n) {
+        buffer.set(next.subarray(0, n - offset), offset)
+        this.#buffers[0] = next.subarray(n - offset)
+        break
+      } else {
+        buffer.set(this.#buffers.shift(), offset)
+        offset += next.length
+      }
+    }
+
+    this.#byteOffset -= n
+
+    return buffer
+  }
+
+  /**
+   * @param {Buffer} buffer
+   */
+  #parseRawHeaders (buffer) {
+    // https://www.rfc-editor.org/rfc/rfc7578#section-4.8
+
+    const headers = this.#info.headers
+    const attributes = this.#info.headers.attributes
+
+    if (buffer.length > maxHeaderLength) {
+      this.#info.skipPart = 'malformed part header'
+      this.emit('error', new Error('Malformed part header'))
+
+      return { headers, attributes }
+    }
+
+    const position = { position: 0 }
+
+    while (position.position < buffer.length) {
+      const nameBuffer = collectASequenceOfCodePoints(
+        (char) => char !== chars[':'],
+        buffer,
+        position
+      )
+
+      if (position.position >= buffer.length) {
+        // Invalid header; doesn't contain a delimiter
+        break
+      } else if (nameBuffer.length === 0) {
+        // https://www.rfc-editor.org/rfc/rfc9110.html#section-5.1-2
+        // field-name     = token
+        // tchar = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+" / "-" / "." /
+        // "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
+        // token = 1*tchar
+
+        // The field name MUST be at least 1 character long.
+        this.#info.skipPart = 'malformed part header'
+        this.emit('error', new Error('Malformed part header'))
+        break
+      }
+
+      // MIME header fields in multipart bodies are required to
+      // consist only of 7-bit data in the US-ASCII character set
+      const name = nameBuffer.toString(this.#opts.defCharset).toLowerCase()
+
+      if (
+        name !== 'content-type' &&
+        name !== 'content-disposition' &&
+        name !== 'content-transfer-encoding'
+      ) {
+        // The multipart/form-data media type does not support any MIME header
+        // fields in parts other than Content-Type, Content-Disposition, and (in
+        // limited circumstances) Content-Transfer-Encoding.  Other header
+        // fields MUST NOT be included and MUST be ignored.
+
+        // Go to the next header if one is available. Headers are split by \r\n,
+        // so we skip all characters until the sequence is found.
+        collectASequenceOfCodePoints(
+          (char) => char !== chars.cr && buffer[position.position + 1] !== chars.lf,
```

Comment:
feels like here should be `||`
maybe rewrite to `!(char === char.cr && buffer[position.position +1] === char.lf)`
583 line same

Reply:
Sorry for being noisy.
Is this code is intended to skip until meet CR **or** LF?
"blablaCR actual finish here CRLF" => skip only to first CR position.

Reply:
It looks for CRLF and returns everything before that